### PR TITLE
Added documentation for DatabaseUrlDataSource

### PR DESCRIPTION
**Pull request by jkutner**
I did not know how to generate the HTML from the `.rst` files, so i apologize if there are mistakes here.

### CODE REVIEW

**Comment by jkutner** [comment]
@szeiger any more revisions needed on this?

File: slick/src/sphinx/database.rst [hunk 1]
```diff
@@ -38,6 +38,27 @@ until the JVM ends (``DB_CLOSE_DELAY=-1``, which is H2 specific).
...
+You can use a Database URL in Typesafe Config as shown here:
+
+.. includecode:: resources/application.conf#dburl
+
+By default, the data source will use the value of the ``DATABASE_URL`` environment variable.
+Thus you may omit the ``url`` property if the ``DATABASE_URL`` environment variable
+is set. You may also define a custom environment variable with standard
+Typesafe Config syntax, such as ``${?MYSQL_DATABASE_URL}``.
+
+Or you may pass a :javaapi:`DatabaseUrlDataSource <slick/jdbc/DatabaseUrlDataSource>` object to
```
Comment (szeiger) [inline slick/src/sphinx/database.rst]: This should be an `:api:` link, not `:javaapi:`

**Comment by szeiger** [comment]
You can test this locally by running `sbt makeSite` for the manual, `sbt doc` for the API docs, or `sbt sdlc` (recommended) for building both and validating the Scaladoc links in the manual. http://sphinx-doc.org/ must be installed for building the manual.

`sbt doctest:test` runs the tests for doc code snippets.

**Comment by jkutner** [comment]
@szeiger Fixed. I can submit another PR to the readme with instructions on how to build the docs (if it's not somewhere else).

**Comment by szeiger** [comment]
Thanks, LGTM. I'll merge and back-port. We want to add a chapter on working on Slick itself. That's probably the best place for documentation about how to build the docs. See #431.

**Pull request merged**